framework/db/QueryBuilder.php

```diff
@@ -1202,13 +1203,13 @@ protected function buildCompositeInCondition($operator, $columns, $values, &$par
             }
             $vss[] = '(' . implode(', ', $vs) . ')';
         }
+
+        $sqlColumns = [];
         foreach ($columns as $i => $column) {
-            if (strpos($column, '(') === false) {
-                $columns[$i] = $this->db->quoteColumnName($column);
-            }
+            $sqlColumns[] = strpos($column, '(') === false ? $this->db->quoteColumnName($column) : $column;
```

Comment:
Ah, my intention was to not call `ArrayHelper::toArray()` and only iterate the values once regardless of it being an array or `\Traversable`.